winch/codegen/src/isa/x64/masm.rs

```diff
@@ -44,11 +44,24 @@ impl Masm for MacroAssembler {
             .mov_rr(stack_pointer, frame_pointer, OperandSize::S64);
     }
 
-    fn push(&mut self, reg: Reg) -> u32 {
-        self.asm.push_r(reg);
-        let increment = <Self::ABI as ABI>::word_bytes();
-        self.increment_sp(increment);
-        self.sp_offset
+    fn push(&mut self, reg: Reg, size: OperandSize) -> StackSlot {
+        let bytes = if reg.is_int() {
+            self.asm.push_r(reg);
+            let increment = <Self::ABI as ABI>::word_bytes();
+            self.increment_sp(increment);
+            increment
+        } else {
+            let bytes = size.bytes();
```

Comment:
I'm wondering now, with type-based spills, if this has the potential to misalign the stack given that `F32`s take 4 bytes. Could you add a comment here about alignment noting how this works?

Reply:
Fixed!